firebase-perf/e2e-app/src/androidTest/java/com/google/firebase/testing/fireperf/FirebasePerformanceTest.java

```diff
@@ -57,9 +47,17 @@ private Intent getTraceAndNetworkEventIntent() {
    *   - Rate: 100 Events/Minute (https://bityl.co/3ZPL)
    */
   @Test
-  public void waitForBothTracesAndNetworkRequestsBatch() throws InterruptedException {
-    Intent intent = getTraceAndNetworkEventIntent();
-    activityRule.launchActivity(intent);
-    FireperfUtils.blockUntilAllEventsSent();
+  public void waitForBothTracesAndNetworkRequestsBatch()
+      throws ExecutionException, InterruptedException {
+    final List<Future<?>> futureList = new ArrayList<>();
+    ActivityScenario scenario = rule.getScenario();
+    scenario.onActivity(
+        activity -> {
+          futureList.add(FireperfUtils.startTraces(15));
```

Comment:
I prefer 15 to be contained in this test, because other tests may use different numbers. I can make a local variable 15 and make both places in this test use that.